includes/class-wc-payments.php

```diff
@@ -94,13 +94,19 @@ public static function init() {
 		include_once dirname( __FILE__ ) . '/class-logger.php';
 		include_once dirname( __FILE__ ) . '/class-wc-payment-gateway-wcpay.php';
 		include_once dirname( __FILE__ ) . '/class-wc-payments-token-service.php';
-		include_once WCPAY_ABSPATH . 'includes/exceptions/class-wc-payments-intent-authentication-exception.php';
+		include_once dirname( __FILE__ ) . '/exceptions/class-wc-payments-intent-authentication-exception.php';
 
 		self::$account          = new WC_Payments_Account( self::$api_client );
 		self::$customer_service = new WC_Payments_Customer_Service( self::$api_client );
 		self::$token_service    = new WC_Payments_Token_Service( self::$api_client, self::$customer_service );
 
-		self::$gateway = new WC_Payment_Gateway_WCPay( self::$api_client, self::$account, self::$customer_service, self::$token_service );
+		$gateway_class = 'WC_Payment_Gateway_WCPay';
+		if ( class_exists( 'WC_Subscriptions' ) ) {
+			include_once dirname( __FILE__ ) . '/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php';
+			$gateway_class = 'WC_Payment_Gateway_WCPay_Subscriptions_Compat';
```

Comment:
I opted for the same approach as the Stripe extension: A subclass of the gateway class with all the Subscriptions bits and pieces encapsulated there. If anyone can think of a cleaner approach, please go ahead.

Reply:
I like this approach, if only because we can speed up our own work by going for something similar to what Stripe is doing 🙂 .